durable_storage/src/persistence_layer.rs

```diff
@@ -0,0 +1,93 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+//! Implementation of the RISC-V PVM durable storage's persistence layer.
+//!
+//! A persistence layer is tied to a repository on disk, identified by a directory.
+//! Within that directory, the persistence layer needs to be able to perform basic KV operations:
+//! - get
+//! - set
+//! - delete
+//!
+//! As well as repository-level operations:
+//! - new
+//! - clone
+//! - commit (returning the commit hash)
+//! - checkout a specific commit
+
+use std::path::Path;
+
+/// Type alias for a 32-byte hash used for identifying key-value blobs & commits.
+type Hash = [u8; 32];
+
+/// Represents a key-value blob to be stored in the persistence layer.
+///
+/// Invariant: A [`KVBlob`] is a content-addressable blob.
+pub struct KVBlob {
+    /// The key is the blake3 hash of the value.
+    key: Hash,
+
+    /// The addressable content of the blob.
+    value: Vec<u8>,
```

Comment:
Is this value variable in size? Is `Vec` the most appropriate container?

Reply:
rocksdb works with `AsRef<[u8]>` and these operations are not expected to change length. So having a `&'data [u8]` seems like the most fitting choice. Thoughts?